nmoscommon/auth/auth_registrar.py

```diff
@@ -0,0 +1,200 @@
+                    "scope": self.allowed_scope,
+                    "redirect_uri": self.redirect_uri,
+                    "grant_type": self.allowed_grant,
+                    "response_type": self.allowed_response,
+                    "token_endpoint_auth_method": self.auth_method
+                    }
+
+            # Decide how Basic Auth details are retrieved - user input? Retrieved from file?
+            reg_resp = requests.post(registration_href,
+                                     data=data,
+                                     auth=('dannym', 'password'),
+                                     timeout=0.5,
+                                     proxies={'http': ''}
+                                     )
+            reg_resp.raise_for_status()  # Raise error if status !=201
+            return reg_resp.json()
+        except HTTPError as e:
+            print("Unable to Register Client with Auth Server. {}".format(e))
+
+    def initialise(self):
+        try:
+            # Check if credentials file already exists i.e. device is already registered
+            if os.path.isfile(CREDENTIALS_PATH):
+                print("Credentials file already exists. Using existing credentials.")
+                self.client_id, self.client_secret = get_credentials_from_file(CREDENTIALS_PATH)
+            else:
+                print("Registering with Authorization Server...")
+                reg_resp_json = self.send_registration_request()
+                self.write_credentials_to_file(reg_resp_json, CREDENTIALS_PATH)
+            return True
+        except Exception as e:
+            print("Unable to initialise OAuth Client with client credentials. {}".format(e))
+            return False
+
+
+class AuthRequestor(object):
+
+    def __init__(self, auth_registrar=None):
+        self.auth_reg = auth_registrar
+        self.bridge = IppmDNSBridge()
+
+    def _auth_request(self, data):
+        try:
+            href = self.bridge.getHref(MDNS_SERVICE_TYPE)
+            if href == '':
+                raise ValueError("No Services Found of type {}".format(MDNS_SERVICE_TYPE))
+            token_href = urljoin(href, TOKEN_ENDPOINT)
+            client_id, client_secret = get_credentials_from_file(CREDENTIALS_PATH)
+
+            # print('Data is: {}. HREF is: {}. ID is: {}. Secret is: {}'.format(
+            #     data, token_href, client_id, client_secret
+            # ))  # Only for Testing
+
+            token_req = requests.post(token_href,
+                                      data=data,
+                                      auth=(client_id, client_secret),
+                                      timeout=0.5,
+                                      proxies={'http': ''})
+            token_req.raise_for_status()
+            print("Token Request Successful")
+            return token_req.json()
+        except Exception as e:
+            print("Bearer Token Request Failed. {}".format(e))
```

Comment:
Log

Reply:
Yeah I left the print statements in for testing purposes.. I knew it wouldnt be too hard to swap over to the logger once I had it ironed out :)